source/Ship.cpp

```diff
@@ -2393,6 +2393,14 @@ int Ship::Scan()
 			Messages::Add("The " + government->GetName() + " " + Noun() + " \""
 					+ Name() + "\" completed its scan of your outfits.", Messages::Importance::High);
 	}
+
+	// Some governments are provoked when scanned.
+	if(auto gov = target->GetGovernment())
+		if(gov->IsHostileWhenScanned()
+				&& !gov->IsEnemy(government)
+				&& (target->Shields() < .9 || target->Hull() < .9 || !target->GetPersonality().IsForbearing())
+				&& !target->GetPersonality().IsPacifist())
+			result |= ShipEvent::PROVOKE;
```

Comment:
I'm not a big fan of the assignment inside the if statement syntax. I also wouldn't use auto and obscure the fact that `gov` is a pointer.
```suggestion
	const Government *gov = target->GetGovernment();
	if(gov && gov->IsHostileWhenScanned() && !gov->IsEnemy(government)
			&& (target->Shields() < .9 || target->Hull() < .9 || !target->GetPersonality().IsForbearing())
			&& !target->GetPersonality().IsPacifist())
		result |= ShipEvent::PROVOKE;
```